Assignment3-Asyncio/process_csv.py

```diff
@@ -0,0 +1,31 @@
+# Python program that reads a CSV into memory and utilizes asycios to process each line in the CSV
+import csv
+import asyncio
+
+async def display_line(reader_obj, writer):
+    for row in reader_obj:
+        print(row)
+        await write_result(row, writer)
+
+
+# This function taking the id and keeping only that id which is greater than 1000 and savong to a results.csv file
+async def write_result(result, writer):
+    async with asyncio.Lock():   # lock for gracefully write to shared file object
+        res = [id for id in result if id != 'PassengerId' and id != 'Survived' and int(id) > 1000] 
+        writer.writerow(res)
```

Comment:
You're currently writing empty lines to your result file, as well as ignoring the "if they survived" flag.  So if we think of it as "We want everyone that was over 1000 and survived", you can do something like:
`        id = int(result[0])`
`        if id > 1000 and result[1]:`
`            writer.writerow([id])`

This makes it possible to skip the write if there's nothing to write getting rid of the empty lines at the top of your results file, as well as gets rid of the need to have the for loop just to select the value you already have, since you can just turn the id in to a list when passing it to writerow.  Also, python interprets a 0 as False, and a 1 as true, so you can can use them in a conditional as in my if statement.

Reply:
Made the changes, so that the output file only contains those id which is greater than 1000 who survived.